java/ql/test/query-tests/security/CWE-312/AllowBackupEnabledTest.ql

```diff
@@ -0,0 +1,21 @@
+import java
+import semmle.code.xml.AndroidManifest
+import TestUtilities.InlineExpectationsTest
+
+class AllowBackupEnabledTest extends InlineExpectationsTest {
+  AllowBackupEnabledTest() { this = "AllowBackupEnabledTest" }
+
+  override string getARelevantTag() { result = "hasAllowBackupEnabled" }
+
+  override predicate hasActualResult(Location location, string element, string tag, string value) {
+    tag = "hasAllowedBackupEnabled" and
+    exists(AndroidApplicationXmlElement androidAppElem |
+      androidAppElem.allowsBackup() and
+      not androidAppElem.getFile().(AndroidManifestXmlFile).isInBuildDirectory()
```

Comment:
This should have the same logic as the main `.ql` file: https://github.com/github/codeql/blob/dca4cd221a38200d0be54f5911002b4151986269/java/ql/src/Security/CWE/CWE-312/AllowBackupAttributeEnabled.ql#L18-26

Reply:
See my other comment about moving all that into the `allowsBackup` predicate.